pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunction.java

```diff
@@ -145,12 +145,23 @@ public void aggregateGroupBySV(int length, int[] groupKeyArray, GroupByResultHol
   @Override
   public void aggregateGroupByMV(int length, int[][] groupKeysArray, GroupByResultHolder groupByResultHolder,
       Map<ExpressionContext, BlockValSet> blockValSetMap) {
-    if (blockValSetMap.isEmpty() || !blockValSetMap.containsKey(STAR_TREE_COUNT_STAR_EXPRESSION)) {
+    if (blockValSetMap.isEmpty()) {
       for (int i = 0; i < length; i++) {
         for (int groupKey : groupKeysArray[i]) {
           groupByResultHolder.setValueForKey(groupKey, groupByResultHolder.getDoubleResult(groupKey) + 1);
         }
       }
+    } else if (_nullHandlingEnabled) {
```

Comment:
nit: IIRC benchmarks shown that performance is almost the same if we don't check for `_nullHandlingEnabled` and simply call forEachNotNull. But I may be wrong

Reply:
Yep, performance was not the concern here, but the fact that we need to handle non-empty vs empty blocks differently (see my other comment [here](https://github.com/apache/pinot/pull/14071#discussion_r1778020911)).